File: Lecture05_RecyclerView/AfterLecture_Java/app/src/main/java/no/hiof/larseknu/playingwithrecyclerview/MovieDetailFragment.java
package no.hiof.larseknu.playingwithrecyclerview;


import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import no.hiof.larseknu.playingwithrecyclerview.model.Movie;

public class MovieDetailFragment extends Fragment {


    public MovieDetailFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_movie_detail, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // Creates a MovieDetailFragmentArgs based on the bundle
        MovieDetailFragmentArgs arguments = MovieDetailFragmentArgs.fromBundle(getArguments());

        // Retreived the uid for the movie that's going to be displayed
        int movieUid = arguments.getUid();

        // Gets the movie with the uid (just a number in the list in this case)
        Movie movie = Movie.getMovies().get(movieUid);

        // Gets a reference to all the views
        TextView movieTitleTextView = view.findViewById(R.id.movieTitleTextView);
        ImageView moviePosterImageView = view.findViewById(R.id.moviePosterImageView);
        TextView movieDescriptionTextView = view.findViewById(R.id.movieDescriptionTextView);

        // Filles up the views with the movie-information
        movieTitleTextView.setText(movie.getTitle());
        moviePosterImageView.setImageResource(movie.getPosterUrl());
        movieDescriptionTextView.setText(movie.getDescription());
    }
}

File: Lecture05_RecyclerView/AfterLecture_Kotlin/app/src/main/java/no/hiof/larseknu/playingwithrecyclerview/MovieDetailFragment.kt
package no.hiof.larseknu.playingwithrecyclerview


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_movie_detail.*
import kotlinx.android.synthetic.main.movie_list_item.*
import kotlinx.android.synthetic.main.movie_list_item.moviePosterImageView
import kotlinx.android.synthetic.main.movie_list_item.movieTitleTextView
import no.hiof.larseknu.playingwithrecyclerview.model.Movie

/**
 * A simple [Fragment] subclass.
 */
class MovieDetailFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_movie_detail, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Retreives the arguments from the bundle (does some null-check)
        val arguments = arguments?.let { MovieDetailFragmentArgs.fromBundle(it) }

        // Gets the movie with the uid (just a number in the list in this case)
        val movie = Movie.getMovies()[arguments!!.uid]

        // Filles up the views with the movie-information
        movieTitleTextView.text = movie.title
        moviePosterImageView.setImageResource(movie.posterUrl)
        movieDescriptionTextView.text = movie.description
    }
}
